### PR TITLE
Remove k8s.io/apimachinery dependency

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

This PR modifies the `github.DownloadReleaseAssets()`  and `github.downloadAssetsParallel()` functions to remove the call to  `errorUtils.AggregateGoroutines()` in the `k8s.io/apimachinery ` package by implementing the parallel download logic ourselves and wraping the errors instead of relying on the module.

Along with with removing `k8s.io/apimachinery` I've updated some dependencies.

#### Which issue(s) this PR fixes:
Ref: kubernetes-sigs/kubetest2#61

#### Special notes for your reviewer:

The function signatures are left as they were. Only the download code has changed.

#### Does this PR introduce a user-facing change?

```release-note
- Implement parallelism ourselves in `github.DownloadReleaseAssets()` instead of using k8s.io/apimachinery
```

### CODE REVIEW

File: pkg/github/github.go [hunk 1]
```diff
@@ -478,12 +483,20 @@ func (g *GitHub) downloadAssetsParallel(assets []github.ReleaseAsset, owner, rep
...
-
 			return nil
 		})
 	}
 
-	return errorUtils.AggregateGoroutines(funcs...)
+	for i := 0; i < cap(errChan); i++ {
+		if err := <-errChan; err != nil {
+			if finalErr == nil {
+				finalErr = errors.New(err.Error())
```
Comment (saschagrunert) [inline pkg/github/github.go:493]: Any reason we do not want to do:
```suggestion
				finalErr = err
```
Here?

**Review by saschagrunert — COMMENTED** [review]
Changes LGTM, just one nit :point_down:

File: pkg/github/github.go (hunk 1 above)
Comment (puerco) [reply pkg/github/github.go]: Leftover from a previous approach :) done :+1:

File: pkg/github/github.go [hunk 2]
```diff
@@ -439,24 +437,31 @@ func (g *GitHub) DownloadReleaseAssets(owner, repo string, releaseTags []string,
...
+			if err := g.downloadAssetsParallel(assets, owner, repo, releaseDir); err != nil {
+				return errors.Wrapf(err, "downloading assets for %s", releaseTag)
 			}
-
 			return nil
 		})
 	}
 
-	return errorUtils.AggregateGoroutines(funcs...)
+	for i := 0; i < cap(errChan); i++ {
```
Comment (saschagrunert) [inline pkg/github/github.go:448]: Okay, why do we use `cap` here instead of `len`?

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**